docs/architecture-decisions/0011-indexing-commands-and-flows.md

```diff
@@ -0,0 +1,66 @@
+
+		 ping     request and display general info about the Elasticsearch server
+
+     help, h  Shows a list of commands or help for one command
+
+   Index actions:
+
+     create   Create an Elasticsearch index
+
+     delete   Delete an Elasticsearch index
+
+     promote  add alias prod to provided index
+
+     demote   remove alias prod from provided index
+
+     indexes  List Elasticsearch indexes with aliases
+
+	   stats    Stats for provided Index (total records, maybe more?)
+
+
+In order to ensure we always have a production index available, we want to use a known, semi-hardcoded alias value. The proposed fixed value is "production".
+
+As we'll have new sources soon, it will be best to have each source maintain their own indexes for ease of use. Elasticsearch will allow multiple indexes to be aliased to "production" to allow us to search as many sources as we need. For aleph, the prefix will be `aleph_`. Future sources will declare a prefix that is appropriate as we add them.
+
+We currently default to an index name of `timdex`. We should no longer do that and instead default to different values depending on the command. Some commands will not have a default at all and are detailed below.
+
+`create` should create a new index automatically if one is not passed in that is based on the source and timestamp. So an example might look like, "aleph_201811081344" for an index created at 1:44pm on 11/18. A `prefix` flag will provide the prefix to the command.
+
+In our manual processing mode, we'd allow that to be created as we parse / ingest or create. (Note: fully autonomous thoughts will follow). We still allow specific index values to be passed in as it may be useful for either intervening in a problem in production, or general local development.
+
+Once we are confident the new index is ready, we'd then `promote` that index while `demoting` any existing `aleph_` indexes. `promote` and `demote` are used to signify adding and removing the alias "production" from the indexes in question. `promote` and `demote` will not default to any index value to ensure intent. The `index` flag will be required to inform this commands what index to operate on. Additionally, we will ensure we do not allow demoting `production` from the last `aleph_` index. In other words, we will programmatically ensure that at least one aleph source is always accessible via the `production` alias.
+
+With this process, we could keep as many old indexes as we deem useful and then go back and delete indexes we no longer need. The delete process will ensure that the index has no aliases, or at least no aliases to `production` before proceeding. Additionally, delete will not default to any index. One must be passed in via the `index` flag.
+
+Those set of commands will allow us to fully manually run the pipeline in production. The following is a proposed automatic flow:
+
+```
+mario ingest --prefix aleph --promote auto --demote auto --url esurl --v4 file
```

Comment:
Under what circumstances would we want to add a new index to the production alias without removing the old index? I think having the alias point to two very similar indices would probably result in double results.

Reply:
I agree. I'm working on addressing all of your feedback. Sorry for the delay in responding to some of this.